src/plugins/typebox.ts

```diff
@@ -18,11 +18,19 @@ const schemas: TSchema[] = [];
 /**
  * Register a schema in the Fastify instance.
  * This function needs to be called before the server starts, so the schema will be registered in the Fastify instance.
- * Each schema needs to have defined the `$id` property.
+ * Each schema needs to have a unique `$id` property.
  * @param schema The schema to be registered.
+ * @param options The options for the schema.
  * @returns The schema passed as argument.
  */
-export function registerSchemaAsRef<T extends TSchema>(schema: T): TRef<T> {
+export function registerSchemaAsRef<T extends TSchema>(
+  schema: T,
+  options: { $id: string; title: string },
+): TRef<T> {
+  // Set schema options
+  schema.$id = options.$id;
+  schema.title = options.title;
```

Comment:
Why was it not working to have `$id` and `title` inside schema directly?

Reply:
Firstly because, they're not required in the schema.
Also, because of the `nullable` issue and some other case (B.e : We can not use `Type.Composite([])` with a `Ref`), we're forced to be able to extract the schema from the ref. So we want to avoid a schema to be used at multiple place with an `$id`. The `$id` property should only be present in `components.schemas` in the OpenAPI JSON, and should not be present in `paths`.